runtime error: index out of range

Using version v0.6.0 mate has a panic and bails out after going through  all the hosted zones.  I have set in the env the AVzone i'm running in ( I should also note that my account has many many hosted zones):
```
time="2017-02-24T19:09:59Z" level=warning msg="[Service] The load balancer of service 'default/nginx' does not have any ingress."
time="2017-02-24T19:10:02Z" level=warning msg="Record [name=default-ingress-nginx.c.tropo.com.] could not be created, another record with same name already exists"
time="2017-02-24T19:10:59Z" level=info msg="[Synchronize] Synchronizing DNS entries..."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'default/kubernetes' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'default/nginx' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'default/nginx-default-backend' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'kube-system/heapster' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'kube-system/kube-dns' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'kube-system/kubernetes-dashboard' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'kube-system/tiller-deploy' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'prod/tropo-prod-notify' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'prod/tropo-staging-notify' does not have any ingress."
time="2017-02-24T19:10:59Z" level=warning msg="[Service] The load balancer of service 'staging/portal-us' does not have any ingress."
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2HYAJG80GL2CW"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZBPFURJ5CBXXZ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z3U5DXDE2JFQB3"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z24Q9Z9B69DT3Y"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZVO7JZ181S8PP"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZZI0WA63X1QL8"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2F0Z8YMJZABVW"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2YT75QE4YM7PA"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z24FJJKBGWNZNZ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z280BZILB8S46A"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2PJFZWWALO32Y"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2V1SZE1UM3XUS"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1WU5ZAKYJ5SKT"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1NDNV2CG95EU6"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2VQHQ6RROARJF"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZLKXRU4MZ3FWV"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1ODC3MG3KVTBQ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1NE4UZ5PX1TJZ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1BH8RO9H1WY7N"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z3SPTVRQJSIZ1Q"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2HTZ0UZ2DPU0D"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1IU0KJK9X1MV4"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z28NLODD9ER8U7"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZTDE75RDOL51W"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z35YCR1BQLN8WT"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1QBJKLR9U9WC7"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2NR6709LCQU8I"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z28UPTDO11QURB"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2D2QNU6HGKMN9"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZWP5NYSQRCX6J"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2DXSTPWBRYO1V"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z38992MPYYP8JK"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZUA23MQXYRINE"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2HLZIFS1NILAA"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZNFN88L3WUJZ3"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2ODSIS9EGZLAI"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z19EX3FG80Q9LQ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z3DF833Y9YRKM2"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1JORVE2JVHCCW"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z16ZVSV7K80FVQ"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2EHBXPWAQ790O"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1N71K8UHPS27L"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z1BKZSKVS0FYTC"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/ZUQNC6AS76ZBW"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z2LE2CC6EFUVFW"
time="2017-02-24T19:10:59Z" level=info msg="No changes submitted for zone:  /hostedzone/Z3E5BG8CWR5ADK"
time="2017-02-24T19:11:00Z" level=info msg="No changes submitted for zone:  /hostedzone/ZD3AG9BKN8PR"
time="2017-02-24T19:11:00Z" level=info msg="No changes submitted for zone:  /hostedzone/ZJH1TDUQN1EGL"
time="2017-02-24T19:11:00Z" level=info msg="No changes submitted for zone:  /hostedzone/ZPVWZ3HLH0LDI"
panic: runtime error: index out of range

goroutine 251 [running]:
github.com/zalando-incubator/mate/consumers.(*awsConsumer).recordInfo(0xc4203baaa0, 0xc420639c00, 0xd0, 0xe0, 0xd0)
	/home/master/workspace/jenkins-build-pierone-login-I64OQ2MMYR6UIUVTGU5ETZUNE67KP4G6TRTQCLZGIAVJKZOQVJ7A/mate/_jenkins_build/go/src/github.com/zalando-incubator/mate/consumers/aws.go:269 +0x1b7
github.com/zalando-incubator/mate/consumers.(*awsConsumer).syncPerHostedZone(0xc4203baaa0, 0x0, 0x0, 0x0, 0xc420376de0, 0x1a, 0xc4204a7f98, 0x1)
	/home/master/workspace/jenkins-build-pierone-login-I64OQ2MMYR6UIUVTGU5ETZUNE67KP4G6TRTQCLZGIAVJKZOQVJ7A/mate/_jenkins_build/go/src/github.com/zalando-incubator/mate/consumers/aws.go:101 +0xb8
github.com/zalando-incubator/mate/consumers.(*awsConsumer).Sync.func1(0xc42033ce40, 0xc4203baaa0, 0xc4205fdb00, 0xc42032ca50, 0xa, 0xc420376de0, 0x1a)
	/home/master/workspace/jenkins-build-pierone-login-I64OQ2MMYR6UIUVTGU5ETZUNE67KP4G6TRTQCLZGIAVJKZOQVJ7A/mate/_jenkins_build/go/src/github.com/zalando-incubator/mate/consumers/aws.go:83 +0xc9
created by github.com/zalando-incubator/mate/consumers.(*awsConsumer).Sync
	/home/master/workspace/jenkins-build-pierone-login-I64OQ2MMYR6UIUVTGU5ETZUNE67KP4G6TRTQCLZGIAVJKZOQVJ7A/mate/_jenkins_build/go/src/github.com/zalando-incubator/mate/consumers/aws.go:89 +0x4ed
```

--- a/consumers/aws.go
+++ b/consumers/aws.go
@@ -265,7 +265,7 @@ func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[strin
 	groupIDMap := map[string]string{} //maps dns to group ID
 
 	for _, record := range records {
-		if (aws.StringValue(record.Type)) == "TXT" {
+		if aws.StringValue(record.Type) == "TXT" && len(record.ResourceRecords) > 0 {
 			groupIDMap[aws.StringValue(record.Name)] = aws.StringValue(record.ResourceRecords[0].Value)
 		} else {
 			if _, exist := groupIDMap[aws.StringValue(record.Name)]; !exist {

--- a/consumers/aws.go
+++ b/consumers/aws.go
@@ -269,7 +269,7 @@ func (a *awsConsumer) groupIDInfo(records []*route53.ResourceRecordSet) map[stri
 			if len(record.ResourceRecords) > 0 {
 				groupIDMap[aws.StringValue(record.Name)] = aws.StringValue(record.ResourceRecords[0].Value)
 			} else {
-				log.Errorf("Unexpected response from AWS API, got TXT record with empty resources: %s . Record is excluded from syncing", aws.StringValue(record.Name))
+				log.Errorf("Unexpected response from AWS API, got TXT record with empty resources: %s. Record is excluded from syncing", aws.StringValue(record.Name))
 				groupIDMap[aws.StringValue(record.Name)] = ""
 			}
 		} else {

--- a/consumers/aws.go
+++ b/consumers/aws.go
@@ -260,8 +260,8 @@ func (a *awsConsumer) getAssignedTXTRecordObject(aliasRecord *route53.ResourceRe
 	}
 }
 
-//recordInfo returns the map of record assigned dns to its target and groupID (can be empty)
-func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[string]*pkg.RecordInfo {
+//groupIDInfo builds a map from dns name to its group ID
+func (a *awsConsumer) groupIDInfo(records []*route53.ResourceRecordSet) map[string]string {
 	groupIDMap := map[string]string{} //maps dns to group ID
 
 	for _, record := range records {
@@ -269,7 +269,7 @@ func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[strin
 			if len(record.ResourceRecords) > 0 {
 				groupIDMap[aws.StringValue(record.Name)] = aws.StringValue(record.ResourceRecords[0].Value)
 			} else {
-				log.Errorf("Unexpected response from AWS API, got TXT record with empty resources: %s. Record is excluded from syncing", aws.StringValue(record.Name))
+				log.Errorf("Unexpected response from AWS API, got TXT record with empty resources: %s . Record is excluded from syncing", aws.StringValue(record.Name))
 				groupIDMap[aws.StringValue(record.Name)] = ""
 			}
 		} else {
@@ -278,7 +278,12 @@ func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[strin
 			}
 		}
 	}
+	return groupIDMap
+}
 
+//recordInfo returns the map of record assigned dns to its target and groupID (can be empty string)
+func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[string]*pkg.RecordInfo {
+	groupIDMap := a.groupIDInfo(records)
 	infoMap := map[string]*pkg.RecordInfo{} //maps record DNS to its GroupID (if exists) and Target (LB)
 	for _, record := range records {
 		groupID := groupIDMap[aws.StringValue(record.Name)]

--- a/consumers/aws.go
+++ b/consumers/aws.go
@@ -265,8 +265,13 @@ func (a *awsConsumer) recordInfo(records []*route53.ResourceRecordSet) map[strin
 	groupIDMap := map[string]string{} //maps dns to group ID
 
 	for _, record := range records {
-		if aws.StringValue(record.Type) == "TXT" && len(record.ResourceRecords) > 0 {
-			groupIDMap[aws.StringValue(record.Name)] = aws.StringValue(record.ResourceRecords[0].Value)
+		if aws.StringValue(record.Type) == "TXT" {
+			if len(record.ResourceRecords) > 0 {
+				groupIDMap[aws.StringValue(record.Name)] = aws.StringValue(record.ResourceRecords[0].Value)
+			} else {
+				log.Errorf("Unexpected response from AWS API, got TXT record with empty resources: %s. Record is excluded from syncing", aws.StringValue(record.Name))
+				groupIDMap[aws.StringValue(record.Name)] = ""
+			}
 		} else {
 			if _, exist := groupIDMap[aws.StringValue(record.Name)]; !exist {
 				groupIDMap[aws.StringValue(record.Name)] = ""

--- a/consumers/aws_test.go
+++ b/consumers/aws_test.go
@@ -127,6 +127,84 @@ func sameTargets(lb1, lb2 string) bool {
 	return lb1 == lb2
 }
 
+func TestGroupIDInfo(t *testing.T) {
+	groupID := "test"
+	client := &awsConsumer{
+		groupID: groupID,
+	}
+	records := []*route53.ResourceRecordSet{
+		&route53.ResourceRecordSet{
+			Type: aws.String("A"),
+			Name: aws.String("test.example.com."),
+			AliasTarget: &route53.AliasTarget{
+				DNSName:      aws.String("abc.def.ghi."),
+				HostedZoneId: aws.String("123"),
+			},
+		},
+		&route53.ResourceRecordSet{
+			Type: aws.String("TXT"),
+			Name: aws.String("test.example.com."),
+			ResourceRecords: []*route53.ResourceRecord{
+				&route53.ResourceRecord{
+					Value: aws.String(client.getGroupID()),
+				},
+			},
+		},
+		&route53.ResourceRecordSet{
+			Type:            aws.String("TXT"),
+			Name:            aws.String("unusual-1.example.com."),
+			ResourceRecords: nil,
+		},
+		&route53.ResourceRecordSet{
+			Type:            aws.String("TXT"),
+			Name:            aws.String("unusual-2.example.com."),
+			ResourceRecords: []*route53.ResourceRecord{},
+		},
+	}
+	groupIDInfoMap := client.groupIDInfo(records)
+	if len(groupIDInfoMap) != 3 {
+		t.Errorf("Incorrect record info for %v", records)
+	}
+	if val, exist := groupIDInfoMap["test.example.com."]; !exist {
+		t.Errorf("Incorrect record info for %v", records)
+	} else {
+		if val != client.getGroupID() {
+			t.Errorf("Incorrect record info for %v", records)
+		}
+	}
+	records = []*route53.ResourceRecordSet{
+		&route53.ResourceRecordSet{
+			Type: aws.String("A"),
+			Name: aws.String("test.example.com."),
+			ResourceRecords: []*route53.ResourceRecord{
+				&route53.ResourceRecord{
+					Value: aws.String("54.32.12.32"),
+				},
+			},
+		},
+		&route53.ResourceRecordSet{
+			Type: aws.String("TXT"),
+			Name: aws.String("test.example.com."),
+			ResourceRecords: []*route53.ResourceRecord{
+				&route53.ResourceRecord{
+					Value: aws.String(client.getGroupID()),
+				},
+			},
+		},
+	}
+	groupIDInfoMap = client.groupIDInfo(records)
+	if len(groupIDInfoMap) != 1 {
+		t.Errorf("Incorrect record info for %v", records)
+	}
+	if val, exist := groupIDInfoMap["test.example.com."]; !exist {
+		t.Errorf("Incorrect record info for %v", records)
+	} else {
+		if val != client.getGroupID() {
+			t.Errorf("Incorrect record info for %v", records)
+		}
+	}
+}
+
 func TestRecordInfo(t *testing.T) {
 	groupID := "test"
 	client := &awsConsumer{

--- a/consumers/aws_test.go
+++ b/consumers/aws_test.go
@@ -152,12 +152,17 @@ func TestRecordInfo(t *testing.T) {
 		},
 		&route53.ResourceRecordSet{
 			Type:            aws.String("TXT"),
-			Name:            aws.String("test.example.com."),
+			Name:            aws.String("unusual-1.example.com."),
 			ResourceRecords: nil,
+		},
+		&route53.ResourceRecordSet{
+			Type:            aws.String("TXT"),
+			Name:            aws.String("unusual-2.example.com."),
+			ResourceRecords: []*route53.ResourceRecord{},
 		},
 	}
 	recordInfoMap := client.recordInfo(records)
-	if len(recordInfoMap) != 1 {
+	if len(recordInfoMap) != 3 {
 		t.Errorf("Incorrect record info for %v", records)
 	}
 	if val, exist := recordInfoMap["test.example.com."]; !exist {

--- a/consumers/aws_test.go
+++ b/consumers/aws_test.go
@@ -150,6 +150,11 @@ func TestRecordInfo(t *testing.T) {
 				},
 			},
 		},
+		&route53.ResourceRecordSet{
+			Type:            aws.String("TXT"),
+			Name:            aws.String("test.example.com."),
+			ResourceRecords: nil,
+		},
 	}
 	recordInfoMap := client.recordInfo(records)
 	if len(recordInfoMap) != 1 {

